node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -467,6 +469,12 @@ void converge() {
         if (!node.equals(lastNode)) {
             logChangesToNodeSpec(lastNode, node);
 
+            if (currentRebootGeneration < node.getCurrentRebootGeneration())
+                currentRebootGeneration = node.getCurrentRebootGeneration();
+
+            if (currentRestartGeneration.isPresent() != node.getCurrentRestartGeneration().isPresent())
```

Comment:
How about just testing for !currentRestartGeneration.isPresent() ? If currentRestartGeneration.isPresent() then we don't want to update currentRestartGeneration even if !node.getCurrentRestartGeneration().isPresent().

Reply:
The reason it is done this way is:
If the node loses allocation, wanted restart generation becomes empty, which is `!=` our restart generation. This will force an update which should fail with https://github.com/vespa-engine/vespa/blob/1e26fba94d5b9101a9da9296dc0ee3ca103fc538/node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/NodePatcher.java#L164